corrupt source files? extended character is not valid in an identifier

### System Information


current git on Gentoo Linux ( e34206bbe0c4d8951230f6a6c7f4aac62c8563e8 ) 


### Expected Behavior

it builds

### Steps to Reproduce

checkout
initialize submodules
mkdir build
cmake ..
make -j10


```
In file included from /home/yoshi/src/TrenchBroom/lib/kdl/test/src/result_test.cpp:25:
/home/yoshi/src/TrenchBroom/lib/kdl/include/kdl/result_io.h:29:40: error: extended character   is not valid in an identifier
   29 |         result.visit([&](const auto& x)��{ str << x; });
      |                                        ^
/home/yoshi/src/TrenchBroom/lib/kdl/include/kdl/result_io.h:37:31: error: extended character   is not valid in an identifier
   37 |             [&](const auto& e)��{ str << e; }
      |                               ^
```

--- a/lib/kdl/include/kdl/result_io.h
+++ b/lib/kdl/include/kdl/result_io.h
@@ -26,7 +26,7 @@
 namespace kdl {
     template <typename Value, typename... Errors>
     std::ostream& operator<<(std::ostream& str, const result<Value, Errors...>& result) {
-        result.visit([&](const auto& x) { str << x; });
+        result.visit([&](const auto& x) { str << x; });
         return str;
     }
 
@@ -34,7 +34,7 @@ namespace kdl {
     std::ostream& operator<<(std::ostream& str, const result<void, Errors...>& result) {
         result.visit(kdl::overload {
             []()               { str << "void"; },
-            [&](const auto& e) { str << e; }
+            [&](const auto& e) { str << e; }
         });
         return str;
     }

